### PR TITLE
Feat: Add CircularMarquee atom component

### LINKED ISSUE #66: Suggestion: Add a CircularMarquee component
Draft from a current project

## script
```js
import { Base } from '@studiometa/js-toolkit';
import { damp, transform } from '@studiometa/js-toolkit/utils';

/**
 * @typedef {CircularMarquee & {
 *   $options: {
 *     sensitivity: number
 *   }
 * }} CircularMarqueeInterface
 */

/**
 * CircularMarquee component
 */
export default class CircularMarquee extends Base {
  /**
   * CircularMarquee Configuration
   */
  static config = {
    name: 'CircularMarquee',
    options: {
      sensitivity: {
        type: Number,
        default: 0.1,
      },
    },
  };

  /**
   * Rotate value.
   * @type {number}
   */
  rotate = 0;

  /**
   * Scroll delta Y.
   * @type {number}
   */
  deltaY = 0;

  /**
   * Transform values.
   */
  transform = {
    rotate: 0,
  };

  /**
   * Update delta scroll on scroll.
   *
   * @param   {import('@studiometa/js-toolkit/services/scroll').ScrollServiceProps} props
   * @returns {void}
   */
  scrolled(props) {
    this.deltaY = props.delta.y;
  }

  /**
   * Update values on raf.
   *
   * @this    {CircularMarqueeInterface}
   * @returns {Function}
   */
  ticked() {
    this.rotate -= (Math.abs(this.deltaY) + 1) * this.$options.sensitivity;

    this.transform.rotate = damp(this.rotate, this.transform.rotate, 0.25);

    return () => {
      transform(this.$el, this.transform);
    };
  }
}
```

## Template
```html
{#
/**
 * @file
 * Circular Marquee
 *
 * @param {String} id - *required
 * @param {Number} outer_radius
 * @param {Number} radius
 * @param {Number} cx
 * @param {Number} cy
 * @param {String} content
 * @param {Object} content_attr
 */
#}

{# Outer radius is svg size radius #}
{% set outer_radius = outer_radius|default(250) %}
{% set outer_diameter = outer_radius * 2 %}

{# Inner radius is the circle size inside the svg (need to be smaller to not cut text with svg viewbox) #}
{% set radius = radius|default(220) %}
{% set diameter = radius * 2 %}
{% set cx = cx|default(radius) %}
{% set cy = cy|default(radius) %}
{% set perimeter = 3.1413 * diameter %}

{% set first = 'M ' ~ (outer_radius - cx) ~ ',' ~ outer_radius %}
{% set second = 'a ' ~ radius ~ ',' ~ radius ~ ' 0 1,1 ' ~ diameter ~ ',0' %}
{% set third = 'a ' ~ radius ~ ',' ~ radius ~ ' 0 1,1 -' ~ diameter ~ ',0' %}
{% set d = [first, second, third]|join(' ') %}

{% set default_content_attributes = {
  'xlink:href': '#' ~ id
} %}
{% set content_attributes =
  merge_html_attributes(default_content_attributes, content_attr|default({}))
%}

<div data-component="CircularMarquee">
  <svg xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 {{ outer_diameter }} {{ outer_diameter }}"
    width="{{ outer_diameter }}"
    height="{{ outer_diameter }}"
    class="z-default relative left-1/2 -translate-x-1/2">
    <defs>
      <path d="{{ d }}" id="{{ id }}" />
    </defs>
    <text dy="0" textLength="{{ perimeter }}">
      <textPath {{ html_attributes(content_attributes) }}>
        {{ content|raw }}
      </textPath>
    </text>
  </svg>
</div>
```

## usage
```html
{% include '@ui/atoms/CircularMarquee.twig' with {
    id: 'douceur',
    radius: 220,
    outer_radius: 250,
    content: ' Douceur • Fraîcheur • Douceur • Fraîcheur • Douceur • Fraîcheur • Douceur • ',
    content_attr: {
       class: 'type-h4 leading-none fill-white'
     }
} only %}
```

## Result

Small rotation by default + rotation on scroll. 
Capture d’écran 2022-08-25 à 15 20 51

### PR DESCRIPTION
**Pull request by perruche**
### 🔗 Linked issue
#66 

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

- Add a CircularMarquee Component, a spinning on scroll circular text. This is made using <svg> text capabilities.
- Resolves #66 

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have added tests (if possible).
- [x] I have updated the documentation accordingly.
- [ ] I have updated the changelog.

### CODE REVIEW

File: packages/docs/components/atoms/CircularMarquee/stories/app.twig [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<div style="height: 400vh">
```
Comment (titouanmathis) [inline packages/docs/components/atoms/CircularMarquee/stories/app.twig:1]: **suggestion :** Add some space before and after the container to enable the scroll and hide the overflow to disable horizontal scroll

```twig
<div class="flex items-center justify-center h-48">
  Scroll down
</div>
<div class="h-screen overflow-hidden">
  {% include '@ui/atoms/CircularMarquee/CircularMarquee.twig' with {
    id: 'douceur',
    radius: 120,
    outer_radius: 150,
    content: ' Douceur • Fraîcheur • Douceur • Fraîcheur • Douceur • Fraîcheur • '
  } only %}
</div>
<div class="flex items-center justify-center h-screen">
  Scroll up
</div>
```

File: packages/docs/components/atoms/CircularMarquee/twig-api.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+- Type: `string`
+
+The Marquee id's. Mandatory to have it unique, as it is used inside the `<svg>` tag to make a reference
+
+### `outer_radius`
+
+- Type: `Number`
+- Defaults: `250`
+
+The outer radius of the `<svg>`. Must be gretter than `radius`.
```
Comment (titouanmathis) [inline packages/docs/components/atoms/CircularMarquee/twig-api.md:21]: **chore :** Typo

```suggestion
The outer radius of the `<svg>`. Must be greater than `radius`.
```

File: packages/docs/components/atoms/CircularMarquee/twig-api.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+The text to insert inside the CircularMarquee
+
+### `sensitivity`
+
+- Type: `Number`
+- Defaults: `0.1`
+
+The sensitivity of the CircularMarquee, that control the animation.
+Use a negative value to make it spin clockwise. The higher the value, the fatest the animation will be.
```
Comment (titouanmathis) [inline packages/docs/components/atoms/CircularMarquee/twig-api.md:42]: **chore:** Wording

```suggestion
Use a negative value to make it spin clockwise. The higher the value, the faster the animation will be.
```

File: packages/ui/atoms/CircularMarquee/CircularMarquee.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import { Base, BaseConfig, BaseProps } from '@studiometa/js-toolkit';
+import { damp, transform } from '@studiometa/js-toolkit/utils';
+
+export interface CircularMarqueeProps extends BaseProps {
+  $options: {
+    sensitivity: number;
+  };
+  rotate: number;
+  deltaY: number;
+  transform: object;
```
Comment (titouanmathis) [inline packages/ui/atoms/CircularMarquee/CircularMarquee.ts:10]: **chore :** Remove these properties

Instance properties declared on the class does not need to be repeated here. The `...Props` interface is only necessary for properties from the framework API: `$options`, `$el`, `$refs` and `$children`. 

```suggestion
```

**Review by titouanmathis — CHANGES_REQUESTED** [review]
I love the effect! 👍 

Some minor tweaks before merging.

File: packages/ui/atoms/CircularMarquee/CircularMarquee.ts (hunk 4 above)
Comment (perruche) [reply packages/ui/atoms/CircularMarquee/CircularMarquee.ts:10]: 👍  add doubt about if they should be here or not when writing it

**Review by titouanmathis — APPROVED** [review]

**Pull request merged**